### PR TITLE
Add Java server encryptor

### PR DESCRIPTION
**Pull request by ipetr0v**
This PR adds an interface for the Java server encryptor and Java encryptor tests.

This encryptor could be used for encryption/decryption unit tests.

### CODE REVIEW

**Review by conradgrobler — APPROVED** [review]

File: java/src/main/java/com/google/oak/crypto/ClientEncryptor.java [hunk 1]
```diff
@@ -59,8 +64,22 @@ public class ClientEncryptor implements Encryptor {
    * <https://secg.org/sec1-v2.pdf>
    */
   // TODO(#3642): Implement Java Hybrid Encryption.
-  public ClientEncryptor(final byte[] serializedServerPublicKey) {
-    this.serializedServerPublicKey = serializedServerPublicKey;
+  public static final Result<ClientEncryptor, Exception> create(
+      final byte[] serializedServerPublicKey) {
```
Comment (rbehjati) [inline java/src/main/java/com/google/oak/crypto/ClientEncryptor.java:68]: I think this whole method can be simplified to 

```
    return Hpke.setupBaseSender(serializedServerPublicKey, OAK_HPKE_INFO).map(ClientEncryptor::new);
```

File: java/src/main/java/com/google/oak/crypto/ClientEncryptor.java [hunk 2]
```diff
@@ -74,39 +93,65 @@ public ClientEncryptor(final byte[] serializedServerPublicKey) {
...
+    // Decrypt response.
+    Result<byte[], Exception> openResult =
+        this.senderResponseContext.open(ciphertext, associatedData);
+    if (openResult.isError()) {
+      return Result.error(openResult.error().get());
+    }
+    byte[] plaintext = openResult.success().get();
+
+    // TODO(#3843): Accept unserialized proto messages once we have Java encryption without JNI.
+    return Result.success(new ClientEncryptor.DecryptionResult(plaintext, associatedData));
```
Comment (rbehjati) [inline java/src/main/java/com/google/oak/crypto/ClientEncryptor.java:155]: All this can also be simplified to one line.

**Review by rbehjati — APPROVED** [review]
I think the code using the `Result`  type can be simplified using a more functional style of coding.

File: java/src/main/java/com/google/oak/crypto/ClientEncryptor.java (hunk 2 above)
Comment (ipetr0v) [reply java/src/main/java/com/google/oak/crypto/ClientEncryptor.java:155]: `ClientEncryptor.DecryptionResult` accepts 2 arguments and `this.senderResponseContext.open` returns only 1

File: java/src/main/java/com/google/oak/crypto/ClientEncryptor.java (hunk 1 above)
Comment (ipetr0v) [reply java/src/main/java/com/google/oak/crypto/ClientEncryptor.java:68]: Done

**Pull request merged**